src/sql/parts/profiler/editor/profilerInput.ts

```diff
@@ -63,22 +66,28 @@ export class ProfilerInput extends EditorInput implements IProfilerSession {
 		this._data = new TableDataView<Slick.SlickData>(undefined, searchFn);
 	}
 
-	public set sessionTemplate(template: IProfilerSessionTemplate) {
-		this._sessionTemplate = template;
-		let newColumns = this.sessionTemplate.view.events.reduce<Array<string>>((p, e) => {
-			e.columns.forEach(c => {
-				if (!p.includes(c)) {
-					p.push(c);
-				}
-			});
+	public set viewTemplate(template: IProfilerViewTemplate) {
+		this._data.clear();
+		this._viewTemplate = template;
+		// need to confirm that this actually does go in order
+		let newColumns = this._viewTemplate.columns.reduce<Array<string>>((p, e) => {
+			p.push(e.name);
 			return p;
 		}, []);
-		newColumns.unshift('EventClass');
+		//Should create a new map of events -> column to display under
+		let newMapping: {[event : string]:string} = {};
```

Comment:
Per our formatter you should have spaces in the type definition: `{ [event: string]: string }`. If you run `tsfmt -r <filenameHere>` then it will save the formatted version of the file (you might have to install tsfmt first by running `npm install -g typescript-formatter`)

Reply:
Thanks for those commands, I didn't know there was an auto formatter! That's super helpful